Wip/fix wsclient bug

msherman64 opened the pull request:
the primary goal here was to workaround https://github.com/kubernetes-client/python/issues/928 , and the first commit seems to mostly do that.

second commit is needed to handle cases where deployment creation failed due to bug #1
third commit needed to resolve error when zun starts and tries to clean up error'd containers from #2 

This all feels fairly hacky, but seems to be an improvement....

msherman64 committed:
kubernetes client: fix leaking sockets

see https://github.com/kubernetes-client/python/issues/928

reusing a kubernetes connection for both stream and non-stream
methods results in `ApiExecption(0) Handshake status 200 OK`

attempt to fix it by allocating new objects for each usage.

msherman64 committed:
set container to error state if deployment scaling fails

msherman64 committed:
implement container kill via stop

Mark-Powers reviewed (approved):
No major concerns here, but a few minor comments in the init

msherman64 commented on zun/container/k8s/driver.py:
```diff
@@ -125,25 +126,25 @@ def __init__(self):
+        # self.network_api = zun_network.api(admin_context, self.net_v1())
         k8s_network = network.K8sNetwork()
-        k8s_network.init(admin_context, self.net_v1)
+        k8s_network.init(admin_context, self.net_v1())
```
good point

msherman64 replied:
This is actually not affecting network and volume, as those are zun drivers instead of k8s client ones.

msherman64 committed:
fix container driver function signatures

in many places, the docker and k8s driver adds new arguments to
overridden methods, but this causes errors when the manager calls
the same method with those arguments missing (or present).

Remove duplicate base class methods and update function arg
names for consistency.

msherman64 commented on zun/container/k8s/driver.py:
```diff
@@ -125,25 +126,25 @@ def __init__(self):
+        self.core_v1 = client.CoreV1Api
+        self.apps_v1 = client.AppsV1Api
+        self.custom = client.CustomObjectsApi
+        self.net_v1 = client.NetworkingV1Api
```
I think it's to make it easier to change/modify the api used, for example if a version or wrapper should be changed.
